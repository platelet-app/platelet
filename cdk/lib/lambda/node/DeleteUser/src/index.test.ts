import { jest, expect } from "@jest/globals";

jest.unstable_mockModule("@platelet-app/lambda", () => ({
    request: jest.fn(),
    errorCheck: jest.fn(),
}));

jest.unstable_mockModule("@aws-sdk/client-cognito-identity-provider", () => {
    const mockSend = jest.fn().mockResolvedValue({});
    const MockCognitoIdentityProviderClient = jest.fn(() => ({
        send: mockSend,
    }));
    return {
        CognitoIdentityProviderClient: MockCognitoIdentityProviderClient,
        AdminDisableUserCommand: jest.fn(),
        AdminDeleteUserCommand: jest.fn(),
        mockSend, // Export mockSend to assert on tests
    };
});

jest.unstable_mockModule("@aws-sdk/client-s3", () => {
    const mockSend = jest.fn().mockResolvedValue({});
    const MockS3Client = jest.fn(() => ({
        send: mockSend,
    }));
    return {
        DeleteObjectCommand: jest.fn(),
        S3Client: MockS3Client,
        mockSend, // Export mockSend to assert on tests
    };
});

// must be imported before the handler
const cognito = await import("@aws-sdk/client-cognito-identity-provider");
const s3 = await import("@aws-sdk/client-s3");
const lambda = await import("@platelet-app/lambda");
// import handler
const { handler } = await import("./index.js");

function setupFetchStub(data: any): () => Promise<Response> {
    return function fetchStub(): Promise<Response> {
        return new Promise((resolve) => {
            resolve({
                json: () =>
                    Promise.resolve({
                        data,
                    }),
            } as Response);
        });
    };
}

const fakeUser = {
    getUser: {
        id: "test",
        _version: 10,
        username: "some-username",
        profilePicture: {
            key: "some-key",
            bucket: "some-bucket",
        },
    },
};

describe("DeleteUser", () => {
    test("delete a user", async () => {
        lambda.request
            .mockImplementationOnce(setupFetchStub(fakeUser))
            .mockImplementation(setupFetchStub({}));
        await handler({
            userId: "test",
            retryCount: 1,
        });
        expect(lambda.request.mock.calls).toMatchSnapshot();
        expect(cognito.mockSend).toHaveBeenCalledTimes(2);
        expect(cognito.AdminDisableUserCommand.mock.calls)
            .toMatchInlineSnapshot(`
            [
              [
                {
                  "UserPoolId": "some_pool",
                  "Username": "some-username",
                },
              ],
            ]
        `);
        expect(cognito.AdminDeleteUserCommand.mock.calls)
            .toMatchInlineSnapshot(`
            [
              [
                {
                  "UserPoolId": "some_pool",
                  "Username": "some-username",
                },
              ],
            ]
        `);
        expect(s3.mockSend).toHaveBeenCalledTimes(1);
        expect(s3.DeleteObjectCommand.mock.calls).toMatchInlineSnapshot(`
            [
              [
                {
                  "Bucket": "some-bucket",
                  "Key": "some-key",
                },
              ],
            ]
        `);
    });
});
