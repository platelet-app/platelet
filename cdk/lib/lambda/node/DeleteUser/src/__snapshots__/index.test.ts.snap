// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`DeleteUser delete a user 1`] = `
[
  [
    {
      "query": "
  query GetUser($id: ID!) {
    getUser(id: $id) {
      id
      disabled
      username
      createdAt
      updatedAt
      profilePicture {
          key
          bucket
      }
      _version
      _deleted
      _lastChangedAt
      __typename
    }
  }
",
      "variables": {
        "id": "test",
      },
    },
    "https://api.example.com/graphql",
  ],
  [
    {
      "query": "mutation DeleteUser(
  $input: DeleteUserInput!
  $condition: ModelUserConditionInput
) {
  deleteUser(input: $input, condition: $condition) {
    id
    username
    cognitoId
    tenantId
    isPrimaryAdmin
    contact {
      name
      telephoneNumber
      mobileNumber
      emailAddress
      ward
      line1
      line2
      line3
      town
      county
      state
      country
      postcode
      what3words
      __typename
    }
    displayName
    name
    roles
    dateOfBirth
    riderResponsibility
    possibleRiderResponsibilities {
      nextToken
      startedAt
      __typename
    }
    profilePicture {
      bucket
      key
      region
      __typename
    }
    comments {
      nextToken
      startedAt
      __typename
    }
    assignments {
      nextToken
      startedAt
      __typename
    }
    vehicleAssignments {
      nextToken
      startedAt
      __typename
    }
    createdTasks {
      nextToken
      startedAt
      __typename
    }
    createdLocations {
      nextToken
      startedAt
      __typename
    }
    createdVehicles {
      nextToken
      startedAt
      __typename
    }
    createdScheduledTasks {
      nextToken
      startedAt
      __typename
    }
    disabled
    createdAt
    updatedAt
    _version
    _deleted
    _lastChangedAt
    __typename
  }
}
",
      "variables": {
        "input": {
          "_version": 10,
          "id": "test",
        },
      },
    },
    "https://api.example.com/graphql",
  ],
]
`;
